### ISSUE TITLE
Dialog disappearance may reduce user experience

### ISSUE THREAD

**Issue by Chou95** [billthefarmer/diary#208]
Hi~ I am using your app and I like it ^-^

However, there exist two dialogs will disappear after phone rotation, which I think may reduce user experience.  Although these events may not breakdown the functionalities of this app, I think the dialog should re-open automatically after rotation for better user experience, right?

I think that you can save the UI state of these dialogs by saving the return value of `dialog.isShowing()`. In this way, you can decide whether to re-open these dialogs after the phone rotation and UI recreation. This may improve the user experience, do you think so?

snapshots to illustrate the details of dialog disappearance:
No.1 before rotation:
![NO2-BEFORE](https://user-images.githubusercontent.com/111506060/215252798-977b2764-b72f-4c8b-9d23-a2fb52149da0.png)
No.1 after rotation:
![NO2-AFTER](https://user-images.githubusercontent.com/111506060/215252822-e3048451-d312-4dc5-a403-bb71dfddf6be.png)
--------------------------------------------------
No.2 before rotation:
![NO1-BEFORE](https://user-images.githubusercontent.com/111506060/215252842-6f954731-8d75-4548-b57d-217d2c92da2a.png)
No.2 after rotation:
![NO1-AFTER](https://user-images.githubusercontent.com/111506060/215252845-5321618c-d5ed-41e6-9e66-cf1e2037d52c.png)

**Comment by billthefarmer** [comment]
I notice that you have been trawling through various apps raising spurious issues about dialogs. I have had similar spurious issues raised by university students using scripts to trawl through code. As far as developers are concerned, this is just time wasting. Dialogs are ephemeral, if you rotate the device while using one, you are assumed to be no longer interested.

**Comment by Chou95** [comment]
> I notice that you have been trawling through various apps raising spurious issues about dialogs. I have had similar spurious issues raised by university students using scripts to trawl through code. As far as developers are concerned, this is just time wasting. Dialogs are ephemeral, if you rotate the device while using one, you are assumed to be no longer interested.

Thank you very much for your reply！

I am studying the data loss issues in Android apps because saving UI states is very important (from the Android developers: https://developer.android.com/topic/libraries/architecture/saving-states), and I have implemented a tool to reveal data loss issues automatically. The issues that I propose here come from my detection results. 

I try to report all the previously unknown data loss issues to developers for feedbacks (I and my friends have also report some other issues, such as crash and losing editable text, and some of them have been fixed in their latest versions). In this way, I want to known whether these events are important to developers and whether these failures can be fixed. 

Additionally, though dialogs are ephemeral, some custom dialogs may be complex and have lots of user input fields, which means user may rotate screen for convenient input. And in many other apps, the dialogs do not disappear after rotation, right? And as I propose above, I think that the fix may not be very complex.

So, I am trying to investigate a general phenomenon. If you (and developers of other apps) could give positive(or negative) feedbacks about whether these dialogs could be fixed, it is helpful.
